### PR TITLE
Use `aws-sdk-go-base/tfawserr` in internal packages

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

After all the work on the `aws-sdk-go-base/tfawserr` package that was initiated by the initial versions of this PR it can now be recast as changing the `internal` packages to use this new package instead of implementing private helpers.
I have simplified this PR by not aliasing the `isAWSErr*` functions in `awserr.go` and moving the Plugin SDK `resource` helpers to a separate PR (https://github.com/terraform-providers/terraform-provider-aws/pull/15477).

Once this is merged I will add a technical debt issue to replace the `isAWSErr` calls with calls to functions in `aws-sdk-go-base/tfawserr`.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

An example resource:

```
 n/a
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
* Renamed `internal/errors` to `internal/tfawserror`
* Renamed `tfawserror.IsAWSErr` to `tfawserror.IsAWSErrCodeMessageContains` to better reflect usage

**Comment by gdavison** [comment]
This is great!

What do you think about making the error functions in `github.com/hashicorp/aws-sdk-go-base` the canonical versions? Possibly moving them into their own package. Then we have a single definition everywhere.

**Comment by ewbankkit** [comment]
@gdavison I wasn't sure of the status of `github.com/hashicorp/aws-sdk-go-base` but I now see very recent commits 😄.
Yes, I think this package should live in that repo (in fact there are already an [`awserr.go`](https://github.com/hashicorp/aws-sdk-go-base/blob/master/awserr.go) there).
`IsResourceNotFoundError` and `IsResourceTimeoutError` should really live in `https://github.com/hashicorp/terraform-plugin-sdk`.

**Comment by ewbankkit** [comment]
I submitted https://github.com/hashicorp/aws-sdk-go-base/pull/37 for the `IsAWSErr` functions.
When that is merged I will rework this PR to use those functions.

**Comment by ewbankkit** [comment]
Pulled in hashicorp/aws-sdk-go-base#37 and refactored.
The `resourceerr` package name for the `terraform-plugin-sdk/helper/resource` functions is just a suggestion. Ideally we could upstream these.

**Comment by ewbankkit** [comment]
##### TODO

* Rebase and make any Terraform Plugin SDK v2 changes
* Cherry pick https://github.com/terraform-providers/terraform-provider-aws/pull/14719

**Comment by ewbankkit** [comment]
Removed _WIP_ after merge of #14719.

File: aws/internal/resourceerr/resource.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package resourceerr
```
Comment (ewbankkit) [inline aws/internal/resourceerr/resource.go:1]: Internal package or add functionality to https://github.com/hashicorp/terraform-plugin-sdk?

**Comment by ewbankkit** [comment]
I've ended up naming the functions that act on `terraform-plugin-sdk/v2/helper/resource` errors based on the conditions that a `true` return value represents: `NotFound` and `TimedOut`.
Example usage

```go
	log.Printf("[DEBUG] Waiting for EC2 Transit Gateway (%s) deletion", transitGatewayID)
	_, err := stateConf.WaitForState()

	if tfresource.NotFound(err) {
		return nil
	}
```

```go
	err := resource.Retry(EventualConsistencyTimeout, func() *resource.RetryError {
		_, err := conn.CreateTags(input)

		if tfawserr.ErrCodeContains(err, ".NotFound") {
			return resource.RetryableError(err)
		}

		if err != nil {
			return resource.NonRetryableError(err)
		}

		return nil
	})

	if tfresource.TimedOut(err) {
		_, err = conn.CreateTags(input)
	}
```

**Comment by ewbankkit** [comment]
I also moved `RetryOnAwsCodes` (which wasn't called anywahere) to the `internal/tfresource` package, renamed to `RetryOnAWSErrorCodes` and added a `context.Context` argument (ready for #15090).

**Review by bflad — APPROVED** [review]
Awesome!! 🚀

**Pull request merged**